clients/gateway.go

```diff
@@ -8,23 +8,64 @@ import (
 	"net/http"
 	"net/url"
 	"strconv"
+	"time"
 
 	"github.com/NethermindEth/juno/core/felt"
 )
 
+type Backoff func(prevWait time.Duration) time.Duration
+
 type GatewayClient struct {
-	baseUrl string
+	url     string
+	backoff Backoff
+	maxWait time.Duration
+	minWait time.Duration
+}
+
+type GatewayOption func(*GatewayClient)
+
+func WithBackoff(b Backoff) GatewayOption {
+	return func(c *GatewayClient) {
+		c.backoff = b
+	}
+}
+
+func WithMaxWait(d time.Duration) GatewayOption {
+	return func(c *GatewayClient) {
+		c.maxWait = d
+	}
+}
+
+func WithMinWait(d time.Duration) GatewayOption {
+	return func(c *GatewayClient) {
+		c.minWait = d
+	}
+}
+
```

Comment:
how about this approach? https://github.com/dgraph-io/badger/blob/83298b771573e8778eb98131913a55e6f30eb11c/options.go#L372

so, something like

```
func (c *Gatewayclient) WithBackoff(b Backoff) *GatewayClient
func (c *Gatewayclient) WithMaxWait(d time.Duration) *GatewayClient

NewGatewayClient().WithBackoff().WithMaxWait()
```

Reply:
That's much simpler, thanks!

Reply:
@IronGauntlets That's what I was doing before, but I think I prefer Omer's solution for simplicity. The only potential drawback I can see is that users can change options after creating the gateway, but I don't think that's an issue here.